### PR TITLE
Simple image reader

### PR DESCRIPTION
**Pull request by vincentsarago**
🤷‍♂️

### CODE REVIEW

**Comment by kylebarron** [comment]
It's beautiful!

**Comment by vincentsarago** [comment]
@kylebarron you're talking about the bike, right?

File: rio_tiler/io/image.py [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+    unscale: bool = False,
+    resampling_method: Resampling = "nearest",
+    post_process: Optional[
+        Callable[[numpy.ndarray, numpy.ndarray], DataMaskType]
+    ] = None,
+    nodata=None,
+) -> DataMaskType:
+    """Low level read function.
+
+    Like rio_tiler.reader.read but without WarpedVRT
```
Comment (vincentsarago) [inline rio_tiler/io/image.py:54]: 👆 

Note: `nodata` is not used right now but keep is there to keep consistency with `rio_tiler.reader.read`

File: rio_tiler/io/image.py [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+        indexes = tuple(indexes) + (idx,)
+        if out_shape:
+            out_shape = (len(indexes), height, width)
+
+        data = src_dst.read(
+            indexes=indexes,
+            window=window,
+            out_shape=out_shape,
+            resampling=resampling,
+            boundless=True,
```
Comment (vincentsarago) [inline rio_tiler/io/image.py:82]: we can use `boundless` here (we can't when working with WarpedVRT)

File: rio_tiler/io/image.py [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+        ycoord = self.tile_size * t.y * res
+
+        return morecantile.Coords(xcoord, ycoord)
+
+    def xy_bounds(self, *tile: morecantile.Tile) -> morecantile.BoundingBox:
+        """Return the bounding box of the (x, y, z) tile"""
+        t = _parse_tile_arg(*tile)
+        left, bottom = self._ul(t)
+        right, top = self._ul(morecantile.Tile(t.x + 1, t.y + 1, t.z))
+        return morecantile.BoundingBox(left, bottom, right, top)
```
Comment (vincentsarago) [inline rio_tiler/io/image.py:141]: Fake TileMatrixSet for local image

**Comment by kylebarron** [comment]
> @kylebarron you're talking about the bike, right?

Yes of course 😄

**Comment by vincentsarago** [comment]
new PR over #533

**Pull request closed without merging**